classes/local/execution/logging_context.php

```diff
@@ -62,11 +62,13 @@ public function log($message) {
             return;
         }
 
-        $logstr = 'Engine "' . $this->engine->name . '"';
+        $logstr = "Engine '{$this->engine->name}'";
+        $strlen = min(20, strlen($this->enginestep->name));
+        $dots = str_repeat('.' , 25 - $strlen);
         if (!is_null($this->enginestep)) {
-            $logstr .= ', step "' . $this->enginestep->name . '"';
+            $logstr .= " step {$this->enginestep->name}{$dots}";
         }
-        $logstr .= ': ' . $message;
+        $logstr .= ' ' . $message;
```

Comment:
So it seems passing the `enginestep->name` (which is derived from a magic getter) into any function is the problem.

You can fix this immediate issue by putting it into a variable first
```suggestion
        $name = $this->engine->name;
        $logstr = "Engine '{$name}'";
        $strlen = min(20, strlen($name));
        $dots = str_repeat('.' , 25 - $strlen);
        if (!is_null($this->enginestep)) {
            $logstr .= " step {$name}{$dots}";
        }
        $logstr .= ' ' . $message;
```